bundles/org.openhab.binding.mpd/src/main/java/org/openhab/binding/mpd/internal/protocol/MPDConnectionThread.java

```diff
@@ -223,40 +229,32 @@ private void closeSocket() {
         if (inputStreamReader != null) {
             try {
                 inputStreamReader.close();
-            } catch (IOException e) {
+            } catch (IOException ignore) {
             }
             this.inputStreamReader = null;
         }
 
-        DataOutputStream writer = this.writer;
-        if (writer != null) {
-            try {
-                writer.close();
-            } catch (IOException e) {
-            }
-            this.writer = null;
-        }
-
         Socket socket = this.socket;
         if (socket != null) {
             try {
                 socket.close();
-            } catch (IOException e) {
+            } catch (IOException ignore) {
             }
             this.socket = null;
         }
     }
 
     private void sendCommand(MPDCommand command) throws IOException {
-        logger.trace("send command '{}'", command.asLine());
-        final DataOutputStream writer = this.writer;
-        if (writer != null) {
+        if (logger.isTraceEnabled()) {
+            logger.trace("send command '{}'", command.asLine());
+        }
+        final Socket socket = this.socket;
```

Comment:
It might make sense to fetch the `socket.getOutputStream()` here so that you don't have to fetch it twice or was there a specific reason to do it like this?

Reply:
Or maybe `socket` can be nulled out and `socket. getOutputStream` not, sensible enough :-)

Reply:
I thought of using a local variable for getOutputStream(), but I find the current code is easier to read and circumvents the problem you describe.